File: Java/gestion-commande/src/main/java/com/ism/restaurant/model/Complement.java
package com.ism.restaurant.model;

public class Complement {

    private int id;
    private String nom;
    private TypeComplement type; // FRITE | BOISSON
    private double prix;
    private int stock;
    private String imageUrl;
    private boolean archive;

    public Complement() {}

    public Complement(int id, String nom, TypeComplement type, double prix, int stock, String imageUrl, boolean archive) {
        this.id = id;
        this.nom = nom;
        this.type = type;
        this.prix = prix;
        this.stock = stock;
        this.imageUrl = imageUrl;
        this.archive = archive;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public String getNom() { return nom; }
    public void setNom(String nom) { this.nom = nom; }
    public double getPrix() { return prix; }
    public void setPrix(double prix) { this.prix = prix; }
    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }
    public boolean isArchive() { return archive; }
    public void setArchive(boolean archive) { this.archive = archive; }
    public TypeComplement getType() { return type; }
    public void setType(TypeComplement type) { this.type = type; }
    public int getStock() { return stock; }
    public void setStock(int stock) { this.stock = stock; }

    @Override
    public String toString() {
        return "Complement{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", type=" + type +
                ", prix=" + prix +
                ", stock=" + stock +
                ", imageUrl='" + imageUrl + '\'' +
                ", archive=" + archive +
                '}';
    }
}

File: src/Entity/Complement.php
<?php

namespace App\Entity;

use App\Repository\ComplementRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ComplementRepository::class)]
#[ORM\Table(name: '`complement`')]
class Complement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 150)]
    private ?string $nom = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: 2)]
    private ?string $prix = null;

    #[ORM\Column(length: 50)]
    private ?string $type = null;

    #[ORM\Column]
    private ?int $stock = null;

    /**
     * @var Collection<int, CommandeComplement>
     */
    #[ORM\OneToMany(targetEntity: CommandeComplement::class, mappedBy: 'complement')]
    private Collection $commandeComplements;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $image_url = null;

    #[ORM\Column]
    private ?bool $isArchive = null;

    public function __construct()
    {
        $this->commandeComplements = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix;
    }

    public function setPrix(string $prix): static
    {
        $this->prix = $prix;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getStock(): ?int
    {
        return $this->stock;
    }

    public function setStock(int $stock): static
    {
        $this->stock = $stock;

        return $this;
    }

    /**
     * @return Collection<int, CommandeComplement>
     */
    public function getCommandeComplements(): Collection
    {
        return $this->commandeComplements;
    }

    public function addCommandeComplement(CommandeComplement $commandeComplement): static
    {
        if (!$this->commandeComplements->contains($commandeComplement)) {
            $this->commandeComplements->add($commandeComplement);
            $commandeComplement->setComplement($this);
        }

        return $this;
    }

    public function removeCommandeComplement(CommandeComplement $commandeComplement): static
    {
        if ($this->commandeComplements->removeElement($commandeComplement)) {
            // set the owning side to null (unless already changed)
            if ($commandeComplement->getComplement() === $this) {
                $commandeComplement->setComplement(null);
            }
        }

        return $this;
    }

    public function getImageUrl(): ?string
    {
        return $this->image_url;
    }

    public function setImageUrl(?string $image_url): static
    {
        $this->image_url = $image_url;

        return $this;
    }

    public function isArchive(): ?bool
    {
        return $this->isArchive;
    }

    public function setIsArchive(bool $isArchive): static
    {
        $this->isArchive = $isArchive;

        return $this;
    }
}
